packages/react/src/hooks/actions/useStateMutationAction.ts

```diff
@@ -0,0 +1,37 @@
+import * as React from 'react';
+import { Hub } from '@aws-amplify/core';
+
+import {
+  ACTIONS_CHANNEL,
+  ACTION_STATE_MUTATION_FINISHED,
+  ACTION_STATE_MUTATION_STARTED,
+} from './constants';
+
+type UseStateMutationAction<T> = [T, (newState: T) => void];
+
+export const useStateMutationAction = <T>(
+  initialState: T
+): UseStateMutationAction<T> => {
+  const [state, setState] = React.useState(initialState);
+
+  const setNewState = React.useCallback(
+    (newState: T) => {
+      const oldState = state;
```

Comment:
Would prefer to call this `prevState` rather than `oldState`, which is more in line with React documentation.
```suggestion
      const prevState = state;
```
https://reactjs.org/docs/hooks-reference.html